### ISSUE TITLE
Tracking Issue for future-incompatibility lint XXX

### ISSUE THREAD

**Issue by h5dh2gy26z-sys** [rust-lang/rust#157132]
This is the **tracking issue** for the `YOUR_LINT_NAME_HERE` future-compatibility warning and other related errors. The goal of this page is to describe why this change was made and how you can fix code that is affected by it. It also provides a place to ask questions or register a complaint if you feel the change should not be made. For more information on the policy around future-compatibility warnings, see our [breaking change policy guidelines][guidelines].

[guidelines]: https://rustc-dev-guide.rust-lang.org/bug-fix-procedure.html

### What is the warning for?

*Describe the conditions that trigger the warning.*

### Why was this change made?

*Explain why this change was made. If there is additional context, like an MCP, link it here.*

### Example

```rust
// Include an example here.
```

### Recommendations

*Give some recommendations on how a user can avoid the lint.*

### When will this warning become a hard error?

*If known, describe the future plans. For example, how long you anticipate this being a warning, or if there are other factors that will influence the anticipated closure.*

### Steps

- [x] Implement the lint
- [x] Raise lint level to deny
- [x] Change the lint to report in dependencies
- [x] Switch to a hard error
